libvips/foreign/vipspng.c

```diff
@@ -564,7 +567,7 @@ png2vips_header( Read *read, VipsImage *out )
 		/* Very large numbers of text chunks are used in DoS
 		 * attacks.
 		 */
-		if( num_text > 10 ) {
+		if( !read->unlimited && num_text > 10 ) {
```

Comment:
20 again. Maybe add a `#define MAX_PNG_TEXT_CHUNKS (20)` to `pforeign.h` (the library-private `foreign/` header).